docs/content/usage/cli/Options.md

```diff
@@ -47,3 +47,23 @@ If you are using this option, make sure you proxy all of the following traffic t
 This traffic redirection is only needed as long as lego solves challenges. As soon as you have received your certificates you can deactivate the forwarding.
 
 [^header]: You must ensure that incoming validation requests contains the correct value for the HTTP `Host` header. If you operate lego behind a non-transparent reverse proxy (such as Apache or NGINX), you might need to alter the header field using `--http.proxy-header X-Forwarded-Host`.
+
+## DNS Resolvers and Challenge Verification
+
+When using a DNS challenge provider (via `--dns <name>`), Lego tries to ensure the ACME challenge token is properly setup before instructing the ACME provider to perform the validation.
+
+This involves a few DNS queries to different servers:
+
+1. Determining the apex domain and resolving CNAMES.
```

Comment:
The goal is to find the "zone", it's not really the apex, it can be the 3rd, 4th, 5th, 6th, etc. level

Reply:
Isn't the apex domain the domain containing the SOA record?

Reply:
Depend on the definition of the apex.
We try to find the domain that handles the subdomain, it can be a subdomain.
An SOA record can be placed on a subdomain.
For me, the apex is the ltd + the "first domain", but maybe my own definition is wrong.

Reply:
I hadn't seen your edit.

Reply:
I've reworded the section and added a bit more content ([preview](https://mm.dmke.org/lego-b649657b/usage/cli/options/#dns-resolvers-and-challenge-verification)).